### PR TITLE
keras 2 compatibility and tests

### PR DESCRIPTION
**Pull request by kmader**
making the code keras 2.0 compatible, adding doctests, validating for channels_first and channels_last settings for keras

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -37,14 +81,321 @@ def g_unet(in_ch, out_ch, nf, batch_size=1, is_binary=False, name='unet'):
...
+    Total params: 26,722.0
+    Trainable params: 24,680.0
+    Non-trainable params: 2,042.0
+    _________________________________________________________________
     """
     merge_params = {
         'mode': 'concat',
         'concat_axis': 1
     }
+    if True:
```
Comment (costapt) [inline models.py]: What is the purpose of this if True?

File: models.py [hunk 2]
```diff
@@ -223,10 +656,23 @@ def pix2pix_loss(y_true, y_pred):
...
     pix2pix.get_layer('d').trainable = False
 
     pix2pix.compile(optimizer=opt, loss=pix2pix_loss)
     return pix2pix
+
+
+if __name__ == '__main__':
+    import doctest
+    # noinspection PyUnresolvedReferences
+    from pyqae.dnn import pix2pix
```
Comment (costapt) [inline models.py]: I am sorry, I have no knowledge of this python module. Can you provide me some info about it?

File: models.py [hunk 3]
```diff
@@ -1,33 +1,77 @@
-"""The model definitions."""
```
Comment (costapt) [inline models.py:1]: Thanks a lot for making the code compatible with the new keras version and adding tests to the code :+1:

File: models.py (hunk 1 above)
Comment (kmader) [reply models.py]: I had it in before, but it isn't needed anymore

File: models.py (hunk 2 above)
Comment (kmader) [reply models.py]: Sorry the pyqae.dnn is a typo, it should be removed, the doctest module (https://docs.python.org/2/library/doctest.html) code basically means when you call the models file as ````python models.py``` it runs all of the tests and checks the outputs

File: models.py [hunk 4]
```diff
@@ -91,58 +440,71 @@ def g_unet(in_ch, out_ch, nf, batch_size=1, is_binary=False, name='unet'):
...
-    dconv1 = Deconvolution(nf * 8, (batch_size, nf*8, 2, 2), k=2, s=1)(x)
+    dconv1 = Deconvolution(nf * 8,
+                           get_deconv_shape(batch_size, nf * 8, 2, 2),
+                           k=2, s=1)(x)
     dconv1 = BatchNorm()(dconv1)
     dconv1 = Dropout(0.5)(dconv1)
-    x = merge([dconv1, conv8], **merge_params)
+    try:
+        x = concatenate_layers([dconv1, conv8], **merge_params)
+    except ValueError:
```
Comment (costapt) [inline models.py]: Why do we want to return this model when the concatenation failed? Shouldn't it be better to let the exception flow throw the user? Or, if you are safeguarding from a known problem, maybe you should print a message and then reraise the exception?

File: models.py (hunk 2 above)
Comment (costapt) [reply models.py]: I see. Thanks for the explanation.

File: models.py (hunk 4 above)
Comment (kmader) [reply models.py]: This was for debugging since the error messages from theano and tensorflow are pretty ugly for mismatched concatentations

**Pull request merged**